### PR TITLE
Documentation for Text add-on.

### PR DESCRIPTION
**Pull request by ajdapretnar**

### CODE REVIEW

File: doc/conf.py [hunk 1]
```diff
@@ -105,7 +106,7 @@
 
 # The theme to use for HTML and HTML Help pages.  See the documentation for
 # a list of builtin themes.
-html_theme = 'sphinx_rtd_theme'
+html_theme = 'nature'
```
Comment (kernc) [inline doc/conf.py]: Good choice. :+1:

File: doc/conf.py [hunk 2]
```diff
@@ -264,6 +265,9 @@
 # If true, do not generate a @detailmenu in the "Top" node's menu.
 #texinfo_no_detailmenu = False
 
+# Mapping for external documentation in Sphinx.
+intersphinx_mapping = {'orange3': ('../../orange3/doc/visual-programming/build/html',
+                                   '../../orange3/doc/visual-programming/build/html/objects.inv')}
```
Comment (kernc) [inline doc/conf.py]: These paths work on your computer. Do they work on mine?

Comment (ajdapretnar) [inline doc/conf.py]: I don't have a clue. 🙈 This was actually just a test and probably shouldn't be in the PR as it doesn't work at all. Thanks, Sphinx, truly. 😠

Comment (ajdapretnar) [inline doc/conf.py]: I commented it out now, just to keep an 'example'. Sphinx community wasn't of help at all.

File: doc/widgets/bagofwords.rst [hunk 3]
```diff
@@ -24,9 +24,39 @@ A set of words from corpus.
 Description
```
Comment (nikicc) [inline doc/widgets/bagofwords.rst]: Could you rephrase `A set of words from corpus.` in the `Signal` sections? Maybe something like: A corpus with added BoW features?

File: doc/widgets/bagofwords.rst [hunk 4]
```diff
@@ -24,9 +24,39 @@ A set of words from corpus.
...
 -----------
 
-No description yet for this widget.
+**Bag of Words** model creates a table with word counts for each data instance (document). The count can be either absolute, binary (contains or does not contain) or sublinear (logarithm of the term frequency). Bag of words model is required in combination with :doc:`Word Enrichment <wordenrichment>` and any kind of predictive modelling.
```
Comment (nikicc) [inline doc/widgets/bagofwords.rst]: creates a `table` — might be confused with Orange's Table. Maybe we rather use Corpus? Or rephrase a bit?

Comment (nikicc) [inline doc/widgets/bagofwords.rst]: `and any kind of predictive modelling`. Not completely true. Should we rather say `could be used for predictive modelling`?

File: doc/widgets/bagofwords.rst [hunk 5]
```diff
@@ -24,9 +24,39 @@ A set of words from corpus.
...
 
 Example
 -------
 
-No example yet for this widget.
+In the first example we will simply check how the bag of words model looks like. Load *bookexcerpts.tab* with :doc:`Corpus <corpus>` widget and connect it to **Bag of Words**. Here we kept the defaults - a simple count of term frequencies. Check what the **Bag of Words** outputs with **Data Table**. The final column in white represents term frequencies for each document.
+
+.. figure:: images/Bag-of-Words-Example1.png
+
+In the second example we will try to predict document category. We are still using the **bookexcerpts.tab* data set, which we sent through :doc:`Preprocess Text <preprocesstext>` with default parameters. Then we connected **Preprocess Text** to **Bag of Words** to obtain term frequencies by which we will compute the model.
```
Comment (nikicc) [inline doc/widgets/bagofwords.rst]: Something seem broken here. A lot of bold text, I see `:doc:` and `<preprocesstext>`, ... Please check.

File: doc/widgets/wordenrichment.rst [hunk 6]
```diff
@@ -26,9 +26,30 @@ Selected instances from corpus.
 Description
 -----------
 
-No description yet for this widget.
+**Word Enrichment** displays a list of words with higher p-values (higher significance) for a
```
Comment (nikicc) [inline doc/widgets/wordenrichment.rst]: Did you mean **lower** p-vales?

File: doc/widgets/wordenrichment.rst [hunk 7]
```diff
@@ -26,9 +26,30 @@ Selected instances from corpus.
...
+selected subset compared to the entire corpus. Lower p-value indicates a higher likelihood that
+the word is significant for the selected subset (not randomly occurring in a text). FDR (False
+Discovery Rate) is linked to p-value and reports on the expected percent of false predictions 
+in the set of predictions, meaning it account for false positives in list of low p-values.
+
+.. figure:: images/Word-Enrichment-stamped.png
+
+1. Information on the input.
+	- Cluster words are all the tokens from the corpus.
```
Comment (nikicc) [inline doc/widgets/wordenrichment.rst]: Could we unify this with the names we used for inputs? We use Corpus here and Data above. Similarly selected subset here and selected data above.

File: doc/widgets/wordcloud.rst [hunk 8]
```diff
@@ -23,14 +23,35 @@ Corpus instance.
...
 Description
 -----------
 
-No description yet for this widget.
+**Word Cloud** displays tokens in the corpus, their size denoting the frequency of the word in corpus. Words are listed by their frequency (weight) in the widget. The widget outputs documents, containing selected tokens from the word cloud.
+
+.. figure:: images/Word-Cloud-stamped.png
+
+1. Information on the input.
+	- number of words(tokens) in a topic
```
Comment (nikicc) [inline doc/widgets/wordcloud.rst]: Add a space between `words` and `(tokens)`.

File: doc/widgets/geomap.rst [hunk 9]
```diff
@@ -13,20 +13,36 @@ Signals
...
 -  **Corpus**
 
-Corpus instance.
+   A :ref:`Corpus` instance.
 
 Description
 -----------
 
-No description yet for this widget.
+**GeoMap** widget shows geolocations from textual (string) data. It finds mentions of geographic names (countries and capitals) and displays distributions (frequency of mentiones) of these names on a map. It works with any Orange widget that outputs a data table and that contains at least one meta attribute. The widget outputs selected data instances, that is all documents containing mentions of a selected country (or countries).
```
Comment (nikicc) [inline doc/widgets/geomap.rst]: `It works with any Orange widget that outputs a data table and that contains at least one meta attribute.` -> It works with any Orange widget that outputs a data table and that contains at least one **string** attribute.

File: doc/widgets/geomap.rst [hunk 10]
```diff
@@ -13,20 +13,36 @@ Signals
...
+2. Select the type of map you wish to display. The options are *World*, *Europe* and *USA*. You can zoom in and out of the map by pressing + and - buttons on a map or by mouse scroll.
+3. The legend for the geographic distribution of data. Countries with the boldest color are most often mentioned in the selected region attribute (highest frequency).
+
+To select documents mentioning a specific country, click on a country and the widget will output matching documents. To select more than one country hold Ctrl/Cmd upon selection.
 
 Example
 -------
 
-No example yet for this widget.
+**GeoMap** widget can be used for simply visualizing distributions of geolocations or for a more complext interactive data analysis. Here, we've queried :doc:`NY Times <nytimes>` for articles on Slovenia for the time period of the last year (2015-2016). First we checked the results with :doc:`Corpus Viewer <corpusviewer>`.
```
Comment (nikicc) [inline doc/widgets/geomap.rst]: Type: `complext`

File: doc/conf.py [hunk 11]
```diff
@@ -310,4 +314,4 @@ def __init__(self, *args, **kwargs):
 # checks if imports work
 from Orange.data import Table
```
Comment (nikicc) [inline doc/conf.py]: Could you comment these three imports out? They were here for testing purposes and I forgot to comment them.

File: doc/widgets/images/Bag-of-Words-tags.txt [hunk 12]
```diff
@@ -0,0 +1,3 @@
+0	253	38
```
Comment (nikicc) [inline doc/widgets/images/Bag-of-Words-tags.txt]: Where is this file used?

File: doc/widgets/images/Corpus-Viewer-tags.txt [hunk 13]
```diff
@@ -0,0 +1,5 @@
+0	157	37
```
Comment (nikicc) [inline doc/widgets/images/Corpus-Viewer-tags.txt]: Same for this one, where is it used?

File: doc/widgets/images/Corpus-tags.txt [hunk 14]
```diff
@@ -0,0 +1,6 @@
+0	206	37
```
Comment (nikicc) [inline doc/widgets/images/Corpus-tags.txt]: Is this used? Same for some files below.

File: doc/widgets/images/Bag-of-Words-tags.txt (hunk 12 above)
Comment (ajdapretnar) [inline doc/widgets/images/Bag-of-Words-tags.txt]: This is generated with stamper.py and it locates number tags on the picture. We had the same discussion with Kernc some time ago and decided to keep these file in the PR. Useful if you simply need to alter the tags a bit and not do them all over again.

File: doc/widgets/nytimes.rst [hunk 15]
```diff
@@ -17,14 +17,40 @@ Signals
...
 -----------
 
-No description yet for this widget.
+**NYTimes** widget loads data from New York Times' Article Search API. You can query NYTimes articles from September
+18, 1851 to today, but the API limit is set to allow retrieving only a 1000 documents per query. You can load up 8 
+attributes per query, with *Headline* and *Lead paragraph* set as default. Publication date, section name and geolocations (appearing in text) are always added to the output.
+
+To use the widget, you must enter `your own API key <https://developer.nytimes.com/signup>`_.
+
+.. figure:: images/NYTimes-stamped.png
```
Comment (ajdapretnar) [inline doc/widgets/nytimes.rst]: 👍

File: doc/widgets/preprocesstext.rst [hunk 16]
```diff
@@ -13,20 +13,89 @@ Signals
...
+
+.. figure:: images/Preprocess-Text-Example1.png
+
+The second example is slightly more complex. We first acquired our data with :doc:`Twitter <twitter>` widget. We quired the internet for tweets from users @HillaryClinton and @realDonaldTrump and got their tweets from the past two weeks, 242 in total.
+
+.. figure:: images/Preprocess-Text-Example2.png
 
-No example yet for this widget.
+In **Preprocess Text** there's *Tweet* tokenization available, which retains hashtags, emojis, mentions and so on. However, this tokenizer doesn't get rid of punctuation, thus we expanded the Regexp filtering with symbols that we wanted to get rid of. We ended up with word-only tokens, which we displayed in :doc:`Word Cloud <wordcloud>`. Then we created a schema for predicting author based on tweet content, which is explained in more details in the documentation for :doc:`Twitter <twitter>` widget.
```
Comment (ajdapretnar) [inline doc/widgets/preprocesstext.rst]: You're right. I think it indeed could be a default, but I can still add it to the documentation. Your call.

File: doc/widgets/wordenrichment.rst (hunk 6 above)
Comment (ajdapretnar) [inline doc/widgets/wordenrichment.rst]: Yes, yes. I meant lower. 🙈

File: doc/widgets/corpus.rst [hunk 17]
```diff
@@ -22,13 +22,36 @@ Corpus instance.
...
+
 Example
 -------
 
-No example yet for this widget.
+The first example shows a very simple use of **Corpus** widget. Place **Corpus** onto canvas and connect
+it to :doc:`Corpus Viewer <corpusviewer>`. We've used *booxexcerpts.tab* data set, which comes with the
+add-on, and inspected it in **Corpus Viewer**.
+
+.. figure:: images/Corpus-Example1.png
```
Comment (ajdapretnar) [inline doc/widgets/corpus.rst]: My bad, my bad.

File: doc/widgets/corpus.rst [hunk 18]
```diff
@@ -22,13 +22,36 @@ Corpus instance.
...
+
+The second example demonstrates how to quickly visualize your corpus with :doc:`Word Cloud <wordcloud>`.
+We could connect **Word Cloud** directly to **Corpus**, but instead we decided to apply some preprocessing
+with :doc:`Preprocess Text <preprocesstext>`. We are again working with *bookexcerpts.tab*. We've
```
Comment (ajdapretnar) [inline doc/widgets/corpus.rst]: It was totally my mistake, I didn't realize I had these loaded. They're my personal data, so we're going with standard (replicable) *bookexcerpts*. :)

File: doc/widgets/preprocesstext.rst (hunk 16 above)
Comment (nikicc) [inline doc/widgets/preprocesstext.rst]: I vote for updating the default value. It could be a part of this PR, just make a separate commit.
